qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -6,646 +6,14 @@
-    local image=$3
-    local snap=$4
-
-    rbd --cluster ${cluster} -p ${pool} snap unprotect ${image}@${snap}
-}
-
-wait_for_snap_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local snap_name=$4
-    local s
-
-    for s in 1 2 4 8 8 8 8 8 8 8 8 16 16 16 16 32 32 32 32; do
-	sleep ${s}
-        rbd --cluster ${cluster} -p ${pool} info ${image}@${snap_name} || continue
-        return 0
-    done
-    return 1
-}
-
-write_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local count=$4
-
-    rbd --cluster ${cluster} -p ${pool} bench-write ${image} \
-	--io-size 4096 --io-threads 1 --io-total $((4096 * count)) \
-	--io-pattern rand
-}
-
-compare_images()
-{
-    local pool=$1
-    local image=$2
-
-    local rmt_export=${TEMPDIR}/${CLUSTER2}-${pool}-${image}.export
-    local loc_export=${TEMPDIR}/${CLUSTER1}-${pool}-${image}.export
-
-    rm -f ${rmt_export} ${loc_export}
-    rbd --cluster ${CLUSTER2} -p ${pool} export ${image} ${rmt_export}
-    rbd --cluster ${CLUSTER1} -p ${pool} export ${image} ${loc_export}
-    cmp ${rmt_export} ${loc_export}
-}
-
-demote_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image demote ${pool}/${image}
-}
-
-promote_image()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image promote ${pool}/${image}
-}
-
-set_pool_mirror_mode()
-{
-    local cluster=$1
-    local pool=$2
-    local mode=$3
-
-    rbd --cluster=${cluster} -p ${pool} mirror pool enable ${mode}
-}
-
-disable_mirror()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image disable ${pool}/${image}
-}
-
-enable_mirror()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-
-    rbd --cluster=${cluster} mirror image enable ${pool}/${image}
-}
-
-test_image_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local test_state=$4
-    local current_state=deleted
-
-    rbd --cluster=${cluster} -p ${pool} ls | grep "^${image}$" &&
-    current_state=present
-
-    test "${test_state}" = "${current_state}"
-}
-
-wait_for_image_present()
-{
-    local cluster=$1
-    local pool=$2
-    local image=$3
-    local state=$4
-    local s
-
-    # TODO: add a way to force rbd-mirror to update replayers
-    for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
-	sleep ${s}
-	test_image_present "${cluster}" "${pool}" "${image}" "${state}" && return 0
-    done
-    return 1
-}
-
-#
-# Main
-#
-
-if [ "$#" -gt 0 ]
-then
-    if [ -z "${RBD_MIRROR_TEMDIR}" ]
-    then
-       echo "RBD_MIRROR_TEMDIR is not set" >&2
-       exit 1
-    fi
-
-    TEMPDIR="${RBD_MIRROR_TEMDIR}"
-    cd ${TEMPDIR}
-    $@
-    exit $?
+if [ -n "${CEPH_REF}" ]; then
+  wget -O rbd_mirror_helpers.sh "https://git.ceph.com/?p=ceph.git;a=blob_plain;hb=$CEPH_REF;f=qa/workunits/rbd/rbd_mirror_helpers.sh"
+  . rbd_mirror_helpers.sh
```

Comment:
@dillaman Sorry, overlooked this before merging to master, but unfortunately this does not work on theuthology:

qa-proxy.ceph.com/teuthology/trociny-2016-06-11_08:56:35-rbd-wip-mgolub-testing---basic-vps/252017/teuthology.log

```
2016-06-11T09:18:53.634 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:--2016-06-11 16:18:53--  https://git.ceph.com/?p=ceph.git;a=blob_plain;hb=769695ee2afa360fecce4a7192fcb7f2f2eb0b25;f=qa/workunits/rbd/rbd_mirror_helpers.sh
2016-06-11T09:18:53.640 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:Resolving git.ceph.com (git.ceph.com)... 8.43.84.132
2016-06-11T09:18:53.641 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:Connecting to git.ceph.com (git.ceph.com)|8.43.84.132|:443... connected.
2016-06-11T09:18:53.826 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:HTTP request sent, awaiting response... 200 OK
2016-06-11T09:18:53.826 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:Length: unspecified [text/x-sh]
2016-06-11T09:18:53.826 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:Saving to: Б
2016-06-11T09:18:53.826 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:
2016-06-11T09:18:53.832 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:     0K .......... ......                                      13.5M=0.001s
2016-06-11T09:18:53.833 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:
2016-06-11T09:18:53.833 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:2016-06-11 16:18:53 (13.5 MB/s) - Б saved [16493]
2016-06-11T09:18:53.833 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:
2016-06-11T09:18:53.833 INFO:tasks.workunit.cluster1.client.mirror.vpm121.stderr:/home/ubuntu/cephtest/workunit.cluster1.client.mirror/rbd/rbd_mirror.sh: 12: .: rbd_mirror_helpers.sh: not found
```
I can't understand why though -- the script is successfully downloaded, and I failed to reproduce locally.